presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -321,7 +321,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     leftPredicate = leftOuterJoinPushDownResult.getOuterJoinPredicate();
```

Comment:
I think it should just be amended to `Push OUTER JIN condition to INNER side in PredicatePushDown`.
I will let you finish the review before doing that.